packages/esm-patient-medications-app/src/order-basket/order-basket-search/drug-search.resource.tsx

```diff
@@ -0,0 +1,122 @@
+import { FetchResponse, openmrsFetch } from '@openmrs/esm-framework';
+import { useMemo } from 'react';
+import useSWRImmutable from 'swr/immutable';
+import { DrugOrderTemplate, OrderTemplate } from '../../api/drug-order-template';
+import { Drug } from '../../types/order';
+import { OrderBasketItem } from '../../types/order-basket-item';
+
+export function useDrugSearch(query): {
+  isLoading: boolean;
+  drugs: Array<Drug>;
+  error: Error;
+} {
+  const { data, error } = useSWRImmutable<FetchResponse<{ results: Array<Drug> }>, Error>(
+    query
+      ? `/ws/rest/v1/drug?q=${query}&v=custom:(uuid,display,name,strength,dosageForm:(display,uuid),concept:(display,uuid))`
+      : null,
+    openmrsFetch,
+  );
+
+  const results = useMemo(
+    () => ({
+      isLoading: !data && !error,
+      drugs: data?.data?.results,
+      error,
+    }),
+    [data, error],
+  );
+
+  return results;
+}
+
+export function useDrugTemplate(drugUuid: string): {
+  isLoading: boolean;
+  templates: Array<DrugOrderTemplate>;
+  error: Error;
+} {
+  const { data, error } = useSWRImmutable<
+    FetchResponse<{
+      results: Array<{
+        uuid: string;
+        drug: Drug;
+        name: string;
+        template: string;
+      }>;
+    }>,
+    Error
+  >(drugUuid ? `/ws/rest/v1/ordertemplates/orderTemplate?drug=${drugUuid}` : null, openmrsFetch);
```

Comment:
Shouldn't we just error out if no `drugUuid` was given?

Reply:
If the drug UUID is not found (i.e. the request to fetch the drugs failed), the component calling this hook will not render. The reason for using the ? : is just for extra-security purpose, but if in any case any error is thrown when fetching the results, it's handled properly.